module-4/backend/week_two.md

```diff
@@ -2,10 +2,62 @@
 
 Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
 
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
-TBD  
+What's one difference between ES5 and ES6?
 
-#### Review  
+ES5 uses function syntax while ES6 uses arrow functions.
 
-TBD
+What's the difference between asynchronous and synchronous JavaScript?
+
+JavaScript is synchronous, meaning it runs on a single thread. However, there are functions available that allow a process to leave the main thread of a Javascript stack and execute an event loop on a separate thread. When the operation returns to the main thread with it's aquired data or completed task, it enters the queue and rejoins the main thread. 
+
+What are the four pillars of Object Oriented programming?
+
+Abstraction - refers to showing only the essential features of the application and hiding the details. 
+Encapsulation - binding data and functions into a class
+Inheritance - Inheritance is a way to reuse code.
+Polymorphism - lets us create functions with same name and different arguments
+
+What are some tools available in JavaScript to help you write object oriented code?
+
+Javascript provides class constructors and the ability to add prototype functions to those classes. 
+
+What are some key concepts to be aware of when refactoring your JavaScript?
+
+Cross-Site Scripting Attack Vulnerability, uncached jQuery selectors
+
+What's a callback function and what are some reasons when we use/need callback functions?
+
+A callback function is a function that is automatically executed when another function is called. We use them when there is some automatic operational response that we want to execute.
+
+What's the scope of variables in Javascript?
+
+Variables are scoped to the function that they exist within. Unless they are global variables. 
+
+What's the difference between == and === in JavaScript?
+
+"===" are strict comparison operators, meaning that both sides of the argument must be identical. 
+"==" is an equality operator that doesn't require both sides to be equal in type, as long as after type conversion they are the same. 
+
+Why do front end frameworks exist?
+
```

Comment:
Good! Frameworks and libraries are similar but different. Think of frameworks as more opinionated about architecture and organization, while libraries standardize a certain domain of code.